FIX: explanation on how to run fMRIPrep

celprov opened the pull request:
fix: add sbatch to run functional workflow of fMRIPrep
fix: indicate the option fMRIPrep in parallel for each session as an alternative if running all sessions in one job is too heavy
enh: precise how to pre-download the template
fix: delete the outdated mention of my fMRIPrep anat-only sbatch

celprov committed:
fix: delete the outdated mention of my fMRIPrep anat-only sbatch
enh: precise how to pre-download the template
fix: correct the script name to launch fmriprep in parallel for each session

celprov committed:
fix: add sbtach to run functional workflow of fMRIPrep
fix: indicate the option fMRIPrep in parallel for each session as an alternative if running all sessions in one job is too heavy

oesteban commented on code/fmriprep/submit-fmriprep.sh:
```diff
@@ -38,7 +38,7 @@
 #     limitations under the License.
 
-DATADIR="/oak/stanford/groups/russpold/inprocess/cprovins/hcph-pilot/"
+DATADIR=<workdir>/data/hcph-dataset
```
You can use secrets and settings here:

```suggestion
DATADIR={{ secrets.data.curnagl_workdir | default('<workdir>') }}/data/hcph-dataset
```

You'll need to make sure you add/confirm the ``curnagl_workdir`` setting in the secrets yaml file.

celprov replied:
I did not want to use secrets here, because we do not render this code in the documentation.
We just point that it exists by telling the user to

> cd code/fmriprep
> bash submit-fmriprep.sh

oesteban replied:
Let's render it, the mkdocs will add a copy icon, so it's way more convenient than having the script on a file.

oesteban commented on code/fmriprep/fmriprep.sbatch:
```diff
@@ -0,0 +1,58 @@
+ml singularityce gcc
+
+WORKDIR={{ secrets.data.curnagl_workdir | default('<workdir>')}}
+SCRATCH=/scratch/cprovins/fmriprep/
```
Perhaps another secret?
```suggestion
SCRATCH={{ secrets.data.curnagl_scratchdir | default('<scratchdir>')}}/fmriprep/
```

oesteban commented on code/fmriprep/fmriprep.sbatch:
```diff
@@ -0,0 +1,58 @@
+#SBATCH --mem=128G 
+#SBATCH --time=24:00:00 
+#SBATCH --export=NONE
+#SBATCH --chdir=/scratch/cprovins
```
Let's make it generic
```suggestion
#SBATCH --chdir=/scratch/%u
```

oesteban commented on code/fmriprep/fmriprep.sbatch:
```diff
@@ -0,0 +1,58 @@
+#
+# General SLURM settings
+#SBATCH --mail-type=ALL 
+#SBATCH --mail-user=<email>@unil.ch
```
Let's create a secret setting:
```suggestion
#SBATCH --mail-user={{ secrets.data.curnagl_email | default('<email>@unil.ch') }}
```

celprov replied:
This does not work as we don't share the same secret (our email address are different).
Let's leave it as such. Also, that's how you put it in fmriprep_anatonly.sbatch

oesteban replied:
I'd prefer to keep an email address. If you don't remember to change the address, at least an email will arrive in your or my inbox :)

oesteban commented on docs/processing/preprocessing.md:
Have you checked whether this is rendering properly?

celprov replied:
Yes I've checked

oesteban reviewed (commented):
A few suggestions :) Thanks for this

celprov committed:
Apply suggestions from code review

Co-authored-by: Oscar Esteban <code@oscaresteban.es>

celprov committed:
Apply suggestions from code review

Co-authored-by: Oscar Esteban <code@oscaresteban.es>

celprov committed:
enh: render submit-fmriprep.sh and ss-fmriprep.sh in the documentation
fix: I had to escape the # in the line defining the number of jobs in submit-fmriprep.sh (penultimate line) and delete the echo line because they made the Markdown integration crash
